### PR TITLE
add the LoggerAwareTrait

### PR DESCRIPTION
**Pull request by franzholz**
It is necessary for the understanding of the code examples.

### CODE REVIEW

**Review by brotkrueml — COMMENTED** [review]
Maybe I am wrong, but the topic is about channels and the added example doesn't set channels. So it is not the same.

**Comment by franzholz** [comment]
This is not an example.
It is the LoggerAwareTrait from the TYPO3 Core file typo3_src-11.5.5/vendor/psr/log/Psr/Log/LoggerAwareTrait.php,

**Comment by brotkrueml** [comment]
You're change request states:

    The same can be accomplished be the use of the :php:`LoggerAwareTrait` :

The code block directly from above uses the security channel, your change not. So it is not the same.

The chapter describes the way to go for TYPO3 v11+ (with dependency injection and attributes)

So, in my opinion, there is not need for giving an example which is not recommended anymore. Maybe others see it different ;-) Just my 2 cents.

**Comment by franzholz** [comment]
The :php:`LoggerAwareInterface`  is not an example.
Who says that is it not recommended any more in TYPO3 11.5?
It is still needed for PHP 7.

File: Documentation/ApiOverview/Logging/Logger/Index.rst [hunk 1]
```diff
@@ -31,6 +31,37 @@ Constructor injection can be used to automatically instantiate the logger:
...
+It is also possible to use the :php:`LoggerAwareInterface`:
+
+.. code-block:: php
+
+   namespace Psr\Log;
+
+   /**
+    * Basic Implementation of LoggerAwareInterface.
+    */
+   trait LoggerAwareTrait
```
Comment (mbrodala) [inline Documentation/ApiOverview/Logging/Logger/Index.rst:43]: It doesn't make sense to embed the trait source here, especially not when the sentence before talks about the `LoggerAwareInterface`.

Instead there should be another `MyClass` example like above using this interface and trait.

**Review by mbrodala — CHANGES_REQUESTED** [review]

**Comment by linawolf** [comment]
@franzholz Are you still working on it? No changes since one month

**Comment by sypets** [comment]
Please look at the other examples on the other pages. LoggerAwareTrait is included in a number of pages:

* [QuickStart](https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/Logging/Quickstart/Index.html#instantiate-a-logger-for-the-current-class)
* [Logger](https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/Logging/Logger/Index.html#channels)

Currently, the examples are inconsistent, sometimes LoggerAwareTrait is used and sometimes dependency injection.

It looks like, DI is now the recommended option (since 11), **not LoggerAwareTrait**.

```
Instantiation
New in version 11.4: [Feature: #95044 - Support autowired LoggerInterface injection](https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/11.4/Feature-95044-SupportAutowiredLoggerInterfaceInjection.html)

Constructor injection can be used to automatically instantiate the logger:

use Psr\Log\LoggerInterface;

class MyClass {
    private LoggerInterface $logger;

    public function __construct(LoggerInterface $logger) {
        $this->logger = $logger;
    }
}
```

* https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/Logging/Logger/Index.html#logger
* see changelog: https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/11.4/Feature-95044-SupportAutowiredLoggerInterfaceInjection.html

I recommend to not merge this PR but ensure that is it generally consistent, and if in doubt, ask in core channel.

**Comment by sypets** [comment]
Please see my PR #1818

**Comment by linawolf** [comment]
Solved with https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/1818

**Pull request closed without merging**